app/hooks/use-display-currency.ts

```diff
@@ -173,6 +179,56 @@ export const useDisplayCurrency = () => {
     [displayCurrencyDictionary],
   )
 
+  const getSecondaryAmountIfCurrencyIsDifferent = useCallback(
+    ({
+      primaryAmount,
+      displayAmount,
+      walletAmount,
+    }: {
+      primaryAmount: MoneyAmount<WalletOrDisplayCurrency>
+      displayAmount: DisplayAmount
+      walletAmount: WalletAmount<WalletCurrency>
+    }) => {
+      // if the display currency is the same as the wallet amount currency, we don't need to show the secondary amount (example: USD display currency with USD wallet amount)
+      if (walletAmount.currency === displayCurrency) {
+        return undefined
+      }
+
+      if (primaryAmount.currency === DisplayCurrency) {
+        return walletAmount
+      }
+
+      return displayAmount
+    },
+    [displayCurrency],
+  )
+
+  const formatDisplayAndWalletAmount = useCallback(
+    ({
+      primaryAmount,
+      displayAmount,
+      walletAmount,
+    }: {
+      primaryAmount?: MoneyAmount<WalletOrDisplayCurrency>
+      displayAmount: DisplayAmount
+      walletAmount: WalletAmount<WalletCurrency>
+    }) => {
+      // if the primary amount is not provided, we use the display amount as the primary amount by default
+      const primaryAmountWithDefault = primaryAmount || displayAmount
```

Comment:
in what scenario would the primary amount not be provided?

Reply:
In some cases you always want it to be formatted `displayAmount (walletAmount)`. If you know that you always want the displayAmount to be first, you can leave out the primaryAmount which is what is used to specify what goes first.